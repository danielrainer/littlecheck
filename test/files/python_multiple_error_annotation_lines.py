# RUN: /usr/bin/python3 %s

# Test that, if error output is reported,
# we get all the remaining lines, not just one.

import sys

sys.stderr.write("stderr1\n")
# CHECKERR: stderr1

sys.stdout.write("fine so far\n")
# CHECK: fine so far

sys.stdout.write("something failed\n")
# That was unexpected, and it would trigger more output on stderr.

sys.stderr.write("stderr2\n")
# CHECKERR: stderr2
sys.stderr.write("stderr3\n")
sys.stderr.write("stderr4\n")
sys.stderr.write("stderr5\n")

sys.stdout.write("now stdout is broken\n")
# CHECK: things should be cool
