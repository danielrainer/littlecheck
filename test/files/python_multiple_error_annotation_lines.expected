Failure in python_multiple_error_annotation_lines.py:

  The CHECK on line 24 wants:
    things should be cool

  which failed to match line stdout:2:
    something failed

  Context:
    fine so far
    something failed <= does not match CHECK on line 24: things should be cool
    now stdout is broken <= no more checks

  There were no remaining checks left to match stderr:3:
    stderr3

  Context:
    stderr1
    stderr2
    stderr3 <= no more checks
    stderr4
    stderr5

  when running command:
    /usr/bin/python3 python_multiple_error_annotation_lines.py
